src/main/java/net/rptools/maptool/model/TokenFootprint.java

```diff
@@ -76,15 +83,33 @@ public String toString() {
     return getLocalizedName();
   }
 
+  public JsonObject toJson() {
+    JsonObject jsonRep = new JsonObject();
+    JsonArray occupiedString = new JsonArray();
+    var cellArray = getOccupiedCells(new CellPoint(0, 0)).toArray();
+    for (int j = 0; j < cellArray.length; j++) {
+      CellPoint currentCell = (CellPoint) cellArray[j];
+      JsonObject jsonPoint = new JsonObject();
+      jsonPoint.addProperty("x", currentCell.x);
+      jsonPoint.addProperty("y", currentCell.y);
+      occupiedString.add(jsonPoint);
+    }
+    jsonRep.addProperty("name", name);
+    jsonRep.add("cells", occupiedString);
+    jsonRep.addProperty("scale", scale);
+    jsonRep.addProperty("localizedName", localizedName);
+    return jsonRep;
+  }
+
```

Comment:
This should be part of the macro function code not here

Reply:
I don't mind either way, but would like to know why

Reply:
Since the functionality is part of the macro and not the footprint I prefer it in the macro code. It's just to stop macro code infiltrating other classes.